fix vitest

@claude

CIが落ちているので、これを直して。エラーログはこれ。liquidからalt-paperに実装を変えました。


RUN v3.2.4 ***/alt-frontend/app

stderr | src/tests/security/automated-scan.test.ts > Automated Security Scan - PROTECTED > Component Security Scan - PROTECTED > should detect potential XSS vulnerabilities in components - PROTECTED
Security vulnerabilities found: [
'/src/components/desktop/timeline/DesktopTimeline.tsx: setTimeout with string detected',
'/src/components/mobile/utils/FloatingMenu.tsx: setTimeout with string detected'
]
Review these issues and fix them if necessary

stderr | src/tests/security/automated-scan.test.ts > Automated Security Scan - PROTECTED > API Security Scan - PROTECTED > should verify CSP compliance in all pages - PROTECTED
CSP compliance issues found: [
'/src/app/desktop/articles/page.tsx: inline style detected - may violate CSP',
'/src/app/desktop/feeds/page.tsx: inline style detected - may violate CSP',
'/src/app/mobile/feeds/_components/ErrorState.tsx: inline event handler detected',
'/src/app/mobile/feeds/favorites/page.tsx: inline style detected - may violate CSP',
'/src/app/mobile/feeds/page.tsx: inline style detected - may violate CSP',
'/src/app/mobile/feeds/viewed/page.tsx: inline style detected - may violate CSP',
'/src/app/page.tsx: inline event handler detected',
'/src/app/test/desktop-sidebar/page.tsx: inline event handler detected'
]

stderr | src/tests/security/automated-scan.test.ts > Automated Security Scan - PROTECTED > Configuration Security Scan - PROTECTED > should detect hardcoded secrets - PROTECTED
Potential hardcoded secrets found: [
'/src/tests/components/desktop/analytics/ReadingAnalytics.test.tsx: potential hardcoded secret detected',
'/src/tests/components/desktop/analytics/RightPanel.test.tsx: potential hardcoded secret detected',
'/src/tests/components/desktop/analytics/TrendingTopics.test.tsx: potential hardcoded secret detected',
'/src/app/desktop/articles/page.tsx: potential hardcoded secret detected',
'/src/app/desktop/feeds/page.tsx: potential hardcoded secret detected',
'/src/app/mobile/articles/search/page.tsx: potential hardcoded secret detected',
'/src/app/test/call-to-action-bar/page.tsx: potential hardcoded secret detected',
'/src/app/test/desktop-layout/page.tsx: potential hardcoded secret detected',
'/src/app/test/desktop-sidebar/page.tsx: potential hardcoded secret detected',
'/src/app/test/quick-actions-panel/page.tsx: potential hardcoded secret detected',
'/src/components/desktop/accessibility/AccessibleDesktopFeedCard.tsx: potential hardcoded secret detected',
'/src/components/desktop/layout/DesktopFeedsLayout.tsx: potential hardcoded secret detected',
'/src/components/desktop/optimized/OptimizedDesktopFeeds.tsx: potential hardcoded secret detected'
]

stderr | src/tests/security/automated-scan.test.ts > Automated Security Scan - PROTECTED > Dependency Security Scan - PROTECTED > should verify no known vulnerable packages - PROTECTED
Potential hardcoded secrets found: [
'next@15.3.3: known vulnerable package detected',
'react@^19.1.1: known vulnerable package detected'
]

stderr | src/tests/security/automated-scan.test.ts > Automated Security Scan - PROTECTED > Code Quality Security Scan - PROTECTED > should verify proper error handling - PROTECTED
Potential hardcoded secrets found: [
'/src/tests/security/automated-scan.test.ts: potential information disclosure in Promise error handling'
]

stderr | src/tests/security/automated-scan.test.ts > Automated Security Scan - PROTECTED > Code Quality Security Scan - PROTECTED > should verify proper type safety - PROTECTED
Potential hardcoded secrets found: [
"/src/tests/theme-integration.test.tsx: unsafe 'any' type usage detected",
'/src/hooks/useFavoriteFeeds.ts: non-null assertion operator usage detected',
'/src/hooks/usePerformanceMonitor.ts: unsafe type assertion detected',
'/src/hooks/useVirtualizationMetrics.ts: unsafe type assertion detected',
'/src/tests/security/regression.test.ts: unsafe type assertion detected',
'/src/utils/dataTransformers.test.ts: unsafe type assertion detected'
]

stderr | src/tests/security/automated-scan.test.ts > Automated Security Scan - PROTECTED > Runtime Security Scan - PROTECTED > should verify no global variable pollution - PROTECTED
Potential hardcoded secrets found: [
'/src/tests/theme-integration.test.tsx: global window object modification detected',
'/src/components/ThemeToggle.tsx: global object modification detected',
'/src/lib/api/auth-client.test.ts: global object modification detected',
'/src/tests/performance/security-impact.test.ts: global object modification detected',
'/src/tests/security/automated-scan.test.ts: global object modification detected',
'/src/tests/security/automated-scan.test.ts: prototype pollution detected',
'/src/tests/security/security.test.ts: global object modification detected'
]

✓ src/tests/security/automated-scan.test.ts (12 tests) 292ms
✓ logic-test/lib/api.test.ts (28 tests) 29ms
✓ logic-test/app/mobile/feeds/stats/errorHandling.test.ts (22 tests) 39ms
✓ src/components/desktop/layout/DesktopSidebar.test.tsx (22 tests) 880ms
✓ src/lib/api/auth-client.test.ts (19 tests) 16ms
✓ src/tests/security/security.test.ts (15 tests) 20ms
✓ src/contexts/auth-context.test.tsx (10 tests) 527ms
✓ logic-test/lib/api-cursor.test.ts (7 tests) 20ms
✓ src/tests/security/regression.test.ts (33 tests) 33ms
✓ src/tests/performance/security-impact.test.ts (14 tests) 403ms
✓ src/components/desktop/timeline/VirtualDesktopTimeline.test.tsx (13 tests) 546ms
✓ logic-test/hooks/useCursorPagination.test.ts (9 tests) 112ms
✓ logic-test/components/VaporwaveBackground.test.ts (21 tests) 14ms
✓ logic-test/utils/cursorPagination.test.ts (9 tests) 12ms
✓ src/utils/errorRecovery.test.ts (17 tests) 9ms
✓ src/utils/featureFlags.test.ts (16 tests) 12ms
✓ logic-test/utils/errorHandler.test.ts (13 tests) 16ms
✓ logic-test/hooks/useSSEFeedsStats.test.ts (12 tests) 74ms
✓ logic-test/schema/validation.test.ts (17 tests) 15ms
✓ logic-test/lib/apiIntegration.test.ts (12 tests) 22ms
✓ src/utils/performanceTracker.test.ts (16 tests) 11ms
✓ logic-test/utils/throttle.test.ts (18 tests) 394ms
✓ src/types/auth.test.ts (12 tests) 7ms
stderr | src/components/mobile/VirtualFeedListImpl.test.tsx > VirtualFeedListImpl > should fallback to fixed sizing when dynamic sizing encounters error
Dynamic sizing error: Error: Test error
at onClick (/alt-frontend/app/src/components/mobile/VirtualFeedListImpl.test.tsx:27:49)
at executeDispatch (/alt-frontend/app/node_modules/.pnpm/react-dom@19.1.1_react@19.1.1/node_modules/react-dom/cjs/react-dom-client.development.js:16368:9)
at runWithFiberInDEV (/alt-frontend/app/node_modules/.pnpm/react-dom@19.1.1_react@19.1.1/node_modules/react-dom/cjs/react-dom-client.development.js:1522:13)
at processDispatchQueue (/alt-frontend/app/node_modules/.pnpm/react-dom@19.1.1_react@19.1.1/node_modules/react-dom/cjs/react-dom-client.development.js:16418:19)
at /alt-frontend/app/node_modules/.pnpm/react-dom@19.1.1_react@19.1.1/node_modules/react-dom/cjs/react-dom-client.development.js:17016:9
at batchedUpdates$1 (/alt-frontend/app/node_modules/.pnpm/react-dom@19.1.1_react@19.1.1/node_modules/react-dom/cjs/react-dom-client.development.js:3262:40)
at dispatchEventForPluginEventSystem (/alt-frontend/app/node_modules/.pnpm/react-dom@19.1.1_react@19.1.1/node_modules/react-dom/cjs/react-dom-client.development.js:16572:7)
at dispatchEvent (/alt-frontend/app/node_modules/.pnpm/react-dom@19.1.1_react@19.1.1/node_modules/react-dom/cjs/react-dom-client.development.js:20658:11)
at dispatchDiscreteEvent (/alt-frontend/app/node_modules/.pnpm/react-dom@19.1.1_react@19.1.1/node_modules/react-dom/cjs/react-dom-client.development.js:20626:11)
at HTMLDivElement.callTheUserObjectsOperation (/alt-frontend/app/node_modules/.pnpm/jsdom@26.1.0/node_modules/jsdom/lib/jsdom/living/generated/EventListener.js:26:30)
An update to VirtualFeedListImpl inside a test was not wrapped in act(...).

When testing, code that causes React state updates should be wrapped into act(...):

act(() => {
/* fire events that update state /
});
/ assert on the output */

This ensures that you're testing the behavior the user would see in the browser. Learn more at https://react.dev/link/wrap-tests-with-act
An update to VirtualFeedListImpl inside a test was not wrapped in act(...).

When testing, code that causes React state updates should be wrapped into act(...):

act(() => {
/* fire events that update state /
});
/ assert on the output */

This ensures that you're testing the behavior the user would see in the browser. Learn more at https://react.dev/link/wrap-tests-with-act

stderr | src/components/mobile/VirtualFeedListImpl.test.tsx > VirtualFeedListImpl > should persist error state after dynamic sizing error
Dynamic sizing error: Error: Test error
at onClick (/alt-frontend/app/src/components/mobile/VirtualFeedListImpl.test.tsx:27:49)
at executeDispatch (/alt-frontend/app/node_modules/.pnpm/react-dom@19.1.1_react@19.1.1/node_modules/react-dom/cjs/react-dom-client.development.js:16368:9)
at runWithFiberInDEV (/alt-frontend/app/node_modules/.pnpm/react-dom@19.1.1_react@19.1.1/node_modules/react-dom/cjs/react-dom-client.development.js:1522:13)
at processDispatchQueue (/alt-frontend/app/node_modules/.pnpm/react-dom@19.1.1_react@19.1.1/node_modules/react-dom/cjs/react-dom-client.development.js:16418:19)
at /alt-frontend/app/node_modules/.pnpm/react-dom@19.1.1_react@19.1.1/node_modules/react-dom/cjs/react-dom-client.development.js:17016:9
at batchedUpdates$1 (/alt-frontend/app/node_modules/.pnpm/react-dom@19.1.1_react@19.1.1/node_modules/react-dom/cjs/react-dom-client.development.js:3262:40)
at dispatchEventForPluginEventSystem (/alt-frontend/app/node_modules/.pnpm/react-dom@19.1.1_react@19.1.1/node_modules/react-dom/cjs/react-dom-client.development.js:16572:7)
at dispatchEvent (/alt-frontend/app/node_modules/.pnpm/react-dom@19.1.1_react@19.1.1/node_modules/react-dom/cjs/react-dom-client.development.js:20658:11)
at dispatchDiscreteEvent (/alt-frontend/app/node_modules/.pnpm/react-dom@19.1.1_react@19.1.1/node_modules/react-dom/cjs/react-dom-client.development.js:20626:11)
at HTMLDivElement.callTheUserObjectsOperation (/alt-frontend/app/node_modules/.pnpm/jsdom@26.1.0/node_modules/jsdom/lib/jsdom/living/generated/EventListener.js:26:30)
An update to VirtualFeedListImpl inside a test was not wrapped in act(...).

When testing, code that causes React state updates should be wrapped into act(...):

act(() => {
/* fire events that update state /
});
/ assert on the output */

This ensures that you're testing the behavior the user would see in the browser. Learn more at https://react.dev/link/wrap-tests-with-act
An update to VirtualFeedListImpl inside a test was not wrapped in act(...).

When testing, code that causes React state updates should be wrapped into act(...):

act(() => {
/* fire events that update state /
});
/ assert on the output */

This ensures that you're testing the behavior the user would see in the browser. Learn more at https://react.dev/link/wrap-tests-with-act

✓ src/components/mobile/VirtualFeedListImpl.test.tsx (7 tests) 181ms
stderr | logic-test/lib/apiSse.test.ts > feedsApiSse > getFeedsStats > should handle invalid JSON messages
Error parsing SSE message: SyntaxError: Unexpected token 'i', "invalid-json" is not valid JSON
at JSON.parse ()
at MockEventSource.eventSource.onmessage (/alt-frontend/app/src/lib/apiSse.ts:14:27)
at MockEventSource.triggerMessage (/alt-frontend/app/logic-test/lib/apiSse.test.ts:40:12)
at /alt-frontend/app/logic-test/lib/apiSse.test.ts:112:32
at file:///alt-frontend/app/node_modules/.pnpm/@vitest+runner@3.2.4/node_modules/@vitest/runner/dist/chunk-hooks.js:155:11
at file:///alt-frontend/app/node_modules/.pnpm/@vitest+runner@3.2.4/node_modules/@vitest/runner/dist/chunk-hooks.js:752:26
at file:///alt-frontend/app/node_modules/.pnpm/@vitest+runner@3.2.4/node_modules/@vitest/runner/dist/chunk-hooks.js:1897:20
at new Promise ()
at runWithTimeout (file:///alt-frontend/app/node_modules/.pnpm/@vitest+runner@3.2.4/node_modules/@vitest/runner/dist/chunk-hooks.js:1863:10)
at runTest (file:///alt-frontend/app/node_modules/.pnpm/@vitest+runner@3.2.4/node_modules/@vitest/runner/dist/chunk-hooks.js:1574:12)

stderr | logic-test/lib/apiSse.test.ts > feedsApiSse > getFeedsStats > should use default error handler when none provided
SSE error

✓ logic-test/lib/apiSse.test.ts (11 tests) 18ms
✓ src/utils/contentSanitizer.test.ts (20 tests) 13ms
✓ src/utils/sizeMeasurement.test.ts (12 tests) 13ms
✓ logic-test/hooks/useReadFeeds.test.ts (7 tests) 350ms
✓ src/components/mobile/DynamicVirtualFeedList.test.tsx (9 tests) 223ms
stderr | logic-test/components/desktop/DesktopFeedCard.test.tsx > DesktopFeedCard > should handle mark as read action
An update to DesktopFeedCard2 inside a test was not wrapped in act(...).

When testing, code that causes React state updates should be wrapped into act(...):

act(() => {
/* fire events that update state /
});
/ assert on the output */

This ensures that you're testing the behavior the user would see in the browser. Learn more at https://react.dev/link/wrap-tests-with-act

stderr | logic-test/components/desktop/DesktopFeedCard.test.tsx > DesktopFeedCard > should handle view article action
An update to DesktopFeedCard2 inside a test was not wrapped in act(...).

When testing, code that causes React state updates should be wrapped into act(...):

act(() => {
/* fire events that update state /
});
/ assert on the output */

This ensures that you're testing the behavior the user would see in the browser. Learn more at https://react.dev/link/wrap-tests-with-act

✓ logic-test/components/desktop/DesktopFeedCard.test.tsx (13 tests) 593ms
✓ logic-test/components/AnimatedNumber.test.ts (17 tests) 21ms
✓ logic-test/lib/api-viewed-feeds.test.ts (8 tests) 13ms
✓ logic-test/utils/infiniteScroll.test.ts (9 tests) 14ms
✓ src/schema/validation/urlValidation.test.ts (10 tests) 20ms
✓ src/components/mobile/VirtualFeedListCore.test.tsx (8 tests) 289ms
✓ src/providers/ThemeProvider.test.tsx (6 tests) 127ms
stderr | logic-test/hooks/useReadFeeds.performance.test.ts > useReadFeeds Performance Tests - PROTECTED > API呼び出し最適化確認 (PROTECTED)
An update to TestComponent inside a test was not wrapped in act(...).

When testing, code that causes React state updates should be wrapped into act(...):

act(() => {
/* fire events that update state /
});
/ assert on the output */

This ensures that you're testing the behavior the user would see in the browser. Learn more at https://react.dev/link/wrap-tests-with-act
An update to TestComponent inside a test was not wrapped in act(...).

When testing, code that causes React state updates should be wrapped into act(...):

act(() => {
/* fire events that update state /
});
/ assert on the output */

This ensures that you're testing the behavior the user would see in the browser. Learn more at https://react.dev/link/wrap-tests-with-act

stderr | logic-test/hooks/useReadFeeds.performance.test.ts > useReadFeeds Performance Tests - PROTECTED > エラー処理の確認 (PROTECTED)
An update to TestComponent inside a test was not wrapped in act(...).

When testing, code that causes React state updates should be wrapped into act(...):

act(() => {
/* fire events that update state /
});
/ assert on the output */

This ensures that you're testing the behavior the user would see in the browser. Learn more at https://react.dev/link/wrap-tests-with-act
An update to TestComponent inside a test was not wrapped in act(...).

When testing, code that causes React state updates should be wrapped into act(...):

act(() => {
/* fire events that update state /
});
/ assert on the output */

This ensures that you're testing the behavior the user would see in the browser. Learn more at https://react.dev/link/wrap-tests-with-act
An update to TestComponent inside a test was not wrapped in act(...).

When testing, code that causes React state updates should be wrapped into act(...):

act(() => {
/* fire events that update state /
});
/ assert on the output */

This ensures that you're testing the behavior the user would see in the browser. Learn more at https://react.dev/link/wrap-tests-with-act
An update to TestComponent inside a test was not wrapped in act(...).

When testing, code that causes React state updates should be wrapped into act(...):

act(() => {
/* fire events that update state /
});
/ assert on the output */

This ensures that you're testing the behavior the user would see in the browser. Learn more at https://react.dev/link/wrap-tests-with-act

stderr | logic-test/hooks/useReadFeeds.performance.test.ts > useReadFeeds Performance Tests - PROTECTED > 先読み機能の動作確認 (PROTECTED)
An update to TestComponent inside a test was not wrapped in act(...).

When testing, code that causes React state updates should be wrapped into act(...):

act(() => {
/* fire events that update state /
});
/ assert on the output */

This ensures that you're testing the behavior the user would see in the browser. Learn more at https://react.dev/link/wrap-tests-with-act
An update to TestComponent inside a test was not wrapped in act(...).

When testing, code that causes React state updates should be wrapped into act(...):

act(() => {
/* fire events that update state /
});
/ assert on the output */

This ensures that you're testing the behavior the user would see in the browser. Learn more at https://react.dev/link/wrap-tests-with-act
An update to TestComponent inside a test was not wrapped in act(...).

When testing, code that causes React state updates should be wrapped into act(...):

act(() => {
/* fire events that update state /
});
/ assert on the output */

This ensures that you're testing the behavior the user would see in the browser. Learn more at https://react.dev/link/wrap-tests-with-act
An update to TestComponent inside a test was not wrapped in act(...).

When testing, code that causes React state updates should be wrapped into act(...):

act(() => {
/* fire events that update state /
});
/ assert on the output */

This ensures that you're testing the behavior the user would see in the browser. Learn more at https://react.dev/link/wrap-tests-with-act
An update to TestComponent inside a test was not wrapped in act(...).

When testing, code that causes React state updates should be wrapped into act(...):

act(() => {
/* fire events that update state /
});
/ assert on the output */

This ensures that you're testing the behavior the user would see in the browser. Learn more at https://react.dev/link/wrap-tests-with-act
An update to TestComponent inside a test was not wrapped in act(...).

When testing, code that causes React state updates should be wrapped into act(...):

act(() => {
/* fire events that update state /
});
/ assert on the output */

This ensures that you're testing the behavior the user would see in the browser. Learn more at https://react.dev/link/wrap-tests-with-act

✓ logic-test/hooks/useReadFeeds.performance.test.ts (4 tests) 997ms
✓ useReadFeeds Performance Tests - PROTECTED > 先読み機能の動作確認 (PROTECTED) 708ms
✓ src/components/mobile/VirtualizedFeedList.test.tsx (5 tests) 47ms
stdout | src/tests/theme-integration.test.tsx > Theme Integration > デフォルトでliquid-beigeテーマが適用される
ThemeToggle mounted, currentTheme: alt-paper

❯ src/tests/theme-integration.test.tsx (1 test | 1 failed) 1036ms
× Theme Integration > デフォルトでliquid-beigeテーマが適用される 1035ms
→ Unable to find an element by: [data-testid="sun-icon"]

Ignored nodes: comments, script, style

Ignored nodes: comments, script, style

stderr | src/hooks/useHomeStats.test.ts > useHomeStats > should return initial loading state An update to TestComponent inside a test was not wrapped in act(...).
When testing, code that causes React state updates should be wrapped into act(...):

act(() => {
/* fire events that update state /
});
/ assert on the output */

This ensures that you're testing the behavior the user would see in the browser. Learn more at https://react.dev/link/wrap-tests-with-act
An update to TestComponent inside a test was not wrapped in act(...).

When testing, code that causes React state updates should be wrapped into act(...):

act(() => {
/* fire events that update state /
});
/ assert on the output */

This ensures that you're testing the behavior the user would see in the browser. Learn more at https://react.dev/link/wrap-tests-with-act

stderr | src/hooks/useHomeStats.test.ts > useHomeStats > should provide refreshStats function
An update to TestComponent inside a test was not wrapped in act(...).

When testing, code that causes React state updates should be wrapped into act(...):

act(() => {
/* fire events that update state /
});
/ assert on the output */

This ensures that you're testing the behavior the user would see in the browser. Learn more at https://react.dev/link/wrap-tests-with-act
An update to TestComponent inside a test was not wrapped in act(...).

When testing, code that causes React state updates should be wrapped into act(...):

act(() => {
/* fire events that update state /
});
/ assert on the output */

This ensures that you're testing the behavior the user would see in the browser. Learn more at https://react.dev/link/wrap-tests-with-act

stderr | src/hooks/useHomeStats.test.ts > useHomeStats > should provide refreshStats function
An update to TestComponent inside a test was not wrapped in act(...).

When testing, code that causes React state updates should be wrapped into act(...):

act(() => {
/* fire events that update state /
});
/ assert on the output */

This ensures that you're testing the behavior the user would see in the browser. Learn more at https://react.dev/link/wrap-tests-with-act
An update to TestComponent inside a test was not wrapped in act(...).

When testing, code that causes React state updates should be wrapped into act(...):

act(() => {
/* fire events that update state /
});
/ assert on the output */

This ensures that you're testing the behavior the user would see in the browser. Learn more at https://react.dev/link/wrap-tests-with-act

✓ src/hooks/useHomeStats.test.ts (5 tests) 238ms
✓ src/utils/dataTransformers.test.ts (5 tests) 6ms
✓ src/utils/searchUtils.test.ts (6 tests) 6ms
stderr | src/hooks/useRecentActivity.test.ts > useRecentActivity > should return initial loading state
An update to TestComponent inside a test was not wrapped in act(...).

When testing, code that causes React state updates should be wrapped into act(...):

act(() => {
/* fire events that update state /
});
/ assert on the output */

This ensures that you're testing the behavior the user would see in the browser. Learn more at https://react.dev/link/wrap-tests-with-act
An update to TestComponent inside a test was not wrapped in act(...).

When testing, code that causes React state updates should be wrapped into act(...):

act(() => {
/* fire events that update state /
});
/ assert on the output */

This ensures that you're testing the behavior the user would see in the browser. Learn more at https://react.dev/link/wrap-tests-with-act
An update to TestComponent inside a test was not wrapped in act(...).

When testing, code that causes React state updates should be wrapped into act(...):

act(() => {
/* fire events that update state /
});
/ assert on the output */

This ensures that you're testing the behavior the user would see in the browser. Learn more at https://react.dev/link/wrap-tests-with-act

✓ src/hooks/useRecentActivity.test.ts (5 tests) 193ms
✓ src/schema/validation/feedUrlSchema.test.ts (5 tests) 16ms
✓ src/utils/linkSafety.test.ts (9 tests) 16ms
✓ src/hooks/useWindowSize.test.ts (5 tests) 43ms
✓ src/hooks/useTheme.test.ts (4 tests) 24ms
✓ logic-test/utils/transform/transformFeedSearch.test.ts (4 tests) 4ms
stdout | src/tests/components/desktop/analytics/RightPanel.test.tsx > RightPanel > should render with glass effect
ThemeToggle mounted, currentTheme: alt-paper

stdout | src/tests/components/desktop/analytics/RightPanel.test.tsx > RightPanel > should show Analytics tab as active by default
ThemeToggle mounted, currentTheme: alt-paper

stdout | src/tests/components/desktop/analytics/RightPanel.test.tsx > RightPanel > should switch between tabs
ThemeToggle mounted, currentTheme: alt-paper

stdout | src/tests/components/desktop/analytics/RightPanel.test.tsx > RightPanel > should use CSS variables for styling
ThemeToggle mounted, currentTheme: alt-paper

✓ src/tests/components/desktop/analytics/RightPanel.test.tsx (4 tests) 563ms
✓ logic-test/components/DesktopHome.test.tsx (1 test) 137ms
✓ src/utils/htmlEscape.test.ts (8 tests) 8ms
✓ src/types/desktop.test.ts (5 tests) 6ms
✓ src/tests/components/desktop/analytics/TrendingTopics.test.tsx (5 tests) 378ms
✓ src/tests/components/desktop/analytics/ReadingAnalytics.test.tsx (5 tests) 350ms
✓ src/tests/security.test.ts (9 tests) 5ms
✓ logic-test/hooks/useTodayUnreadCount.test.ts (2 tests) 124ms
✓ logic-test/components/desktop/layout/DesktopFeedsLayout.test.tsx (2 tests) 4ms
✓ logic-test/utils/time.test.ts (1 test) 3ms

⎯⎯⎯⎯⎯⎯⎯ Failed Tests 1 ⎯⎯⎯⎯⎯⎯⎯

FAIL src/tests/theme-integration.test.tsx > Theme Integration > デフォルトでliquid-beigeテーマが適用される
TestingLibraryElementError: Unable to find an element by: [data-testid="sun-icon"]

Ignored nodes: comments, script, style

<div>
  <div
    data-testid="theme-toggle"
    display="inline-block"
  >
    <button
      aria-checked="false"
      aria-label="Switch to Liquid Beige theme. Current theme: Liquid Beige"
      data-testid="theme-toggle-button"
      role="switch"
      style="width: 40px; height: 40px; min-width: 40px; border-radius: 12px; position: relative; overflow: hidden; transition: all 0.2s cubic-bezier(0.4, 0, 0.2, 1); border: 1px solid var(--surface-border);"
    />
Test Files 1 failed | 57 passed (58)
Tests 1 failed | 603 passed (604)
Start at 10:22:52
Duration 67.76s (transform 891ms, setup 4.41s, collect 15.91s, tests 9.58s, environment 25.71s, prepare 4.45s)

  </div>
</div>
Ignored nodes: comments, script, style

      <button
Error: TestingLibraryElementError: Unable to find an element by: [data-testid="sun-icon"]

Ignored nodes: comments, script, style

Ignored nodes: comments, script, style

❯ Proxy.waitForWrapper node_modules/.pnpm/https://github.com/testing-library+dom@10.4.0/node_modules/@testing-library/dom/dist/wait-for.js:163:27 ❯ src/__tests__/theme-integration.test.tsx:125:11
        aria-checked="false"
        aria-label="Switch to Liquid Beige theme. Current theme: Liquid Beige"
        data-testid="theme-toggle-button"
        role="switch"
        style="width: 40px; height: 40px; min-width: 40px; border-radius: 12px; position: relative; overflow: hidden; transition: all 0.2s cubic-bezier(0.4, 0, 0.2, 1); border: 1px solid var(--surface-border);"
      />
    </div>
  </div>
</body>
❯ Proxy.waitForWrapper node_modules/.pnpm/https://github.com/testing-library+dom@10.4.0/node_modules/@testing-library/dom/dist/wait-for.js:163:27 ❯ src/__tests__/theme-integration.test.tsx:125:11 123| 124| // Wait for the component to fully mount and show icon 125| await waitFor(() => { | ^ 126| expect(screen.getByTestId("sun-icon")).toBeDefined(); 127| });
⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯[1/1]⎯

Error: Process completed with exit code 1.

--- a/alt-frontend/app/src/__tests__/theme-integration.test.tsx
+++ b/alt-frontend/app/src/__tests__/theme-integration.test.tsx
@@ -80,17 +80,17 @@ describe("Theme Integration", () => {
   beforeEach(() => {
     vi.clearAllMocks();
     localStorageMock.getItem.mockReturnValue(null);
-    getAttributeMock.mockReturnValue("liquid-beige");
+    getAttributeMock.mockReturnValue("alt-paper");
 
-    // Setup default CSS variables for liquid-beige theme
+    // Setup default CSS variables for alt-paper theme
     mockGetComputedStyle.mockReturnValue({
       getPropertyValue: vi.fn((prop: string) => {
         const cssVars: Record<string, string> = {
           "--foreground": "#2c2c2c",
-          "--background": "#d1c0a8",
-          "--accent-primary": "#b85450",
-          "--accent-secondary": "#7c9070",
-          "--accent-tertiary": "#d4a574",
+          "--background": "#f8f6f2",
+          "--accent-primary": "#6b7280",
+          "--accent-secondary": "#9ca3af",
+          "--accent-tertiary": "#d1d5db",
         };
         return cssVars[prop] || "";
       }),
@@ -105,7 +105,7 @@ describe("Theme Integration", () => {
     window.getComputedStyle = originalGetComputedStyle;
   });
 
-  it("デフォルトでliquid-beigeテーマが適用される", async () => {
+  it("デフォルトでalt-paperテーマが適用される", async () => {
     render(
       <ThemeProvider>
         <ThemeToggle />
@@ -117,9 +117,9 @@ describe("Theme Integration", () => {
       expect(screen.getByTestId("theme-toggle-button")).toBeDefined();
     });
 
-    // Verify default theme is liquid-beige
+    // Verify default theme is alt-paper
     const toggleButton = screen.getByTestId("theme-toggle-button");
-    expect(toggleButton.getAttribute("aria-checked")).toBe("false"); // liquid-beige = false (not vaporwave)
+    expect(toggleButton.getAttribute("aria-checked")).toBe("false"); // alt-paper = false (not vaporwave)
 
     // Wait for the component to fully mount and show icon
     await waitFor(() => {

--- a/alt-frontend/app/src/components/ThemeToggle.tsx
+++ b/alt-frontend/app/src/components/ThemeToggle.tsx
@@ -19,7 +19,7 @@ export interface ThemeToggleProps {
 
 const THEME_ICONS = {
   vaporwave: Moon,
-  "liquid-beige": Sun,
+  "alt-paper": Sun,
 };
 
 export const ThemeToggle: React.FC<ThemeToggleProps> = ({
@@ -99,11 +99,11 @@ export const ThemeToggle: React.FC<ThemeToggleProps> = ({
 
   const IconComponent = THEME_ICONS[currentTheme];
   const nextTheme =
-    currentTheme === "liquid-beige" ? "vaporwave" : "liquid-beige";
+    currentTheme === "alt-paper" ? "vaporwave" : "alt-paper";
   const nextThemeLabel =
-    nextTheme === "vaporwave" ? "Vaporwave" : "Liquid Beige";
+    nextTheme === "vaporwave" ? "Vaporwave" : "Alt Paper";
   const currentThemeLabel =
-    currentTheme === "vaporwave" ? "Vaporwave" : "Liquid Beige";
+    currentTheme === "vaporwave" ? "Vaporwave" : "Alt Paper";
 
   // Use CSS variables from global.css that work with data-style attribute
   const buttonStyles = {

--- a/alt-frontend/app/src/components/mobile/VirtualFeedListImpl.test.tsx
+++ b/alt-frontend/app/src/components/mobile/VirtualFeedListImpl.test.tsx
@@ -1,6 +1,6 @@
 import React from "react";
 import { describe, it, expect, vi, beforeEach } from "vitest";
-import { render, screen, waitFor } from "@testing-library/react";
+import { render, screen, waitFor, act } from "@testing-library/react";
 import { ChakraProvider, defaultSystem } from "@chakra-ui/react";
 import { VirtualFeedListImpl } from "./VirtualFeedListImpl";
 import { Feed } from "@/schema/feed";
@@ -140,7 +140,9 @@ describe("VirtualFeedListImpl", () => {
 
     // Trigger error
     const errorButton = screen.getByText("Trigger Error");
-    errorButton.click();
+    await act(async () => {
+      errorButton.click();
+    });
 
     // Should fallback to fixed sizing
     await waitFor(() => {
@@ -187,7 +189,9 @@ describe("VirtualFeedListImpl", () => {
 
     // Trigger error
     const errorButton = screen.getByText("Trigger Error");
-    errorButton.click();
+    await act(async () => {
+      errorButton.click();
+    });
 
     // Wait for fallback
     await waitFor(() => {

